artemis-server/src/main/java/org/apache/activemq/artemis/core/server/RoutingContext.java

```diff
@@ -82,6 +82,9 @@
 
    boolean isReusable(Message message, int version);
 
+   /** Provides human readable information about the routing, for tracing and debug */
+   String debug();
```

Comment:
Isn't that just a toString method then?

Reply:
I wanted to keep it separate TBH.

message.toString() may include routing, and that will include too much information.

Reply:
If thats the case does it need to be a method in interface? If during some custom debug, you can just cast the type to access the method. Not sure we' want it a public advertised method.

Reply:
fair enough. I will take it out of the interface.

Reply:
it would look horrible though. on PostOfficeImpl, after routed I want to include the debug information on that case.

This would require a case, and test if RoutingContext implements RoutingContextImpl.

Reply:
I think it's better to keep it.

Reply:
know what.. I will just make it toString()

Reply:
solved